lighthouse-core/test/audits/font-display-test.js

```diff
@@ -5,89 +5,110 @@
+      }
 
-    return Audit.audit(getArtifacts([
+      @font-face {
+        src: url(font.woff);
+      }
+    `;
+
+    networkRecords = [
       {
-        url: openSansFont.src[0],
+        url: 'https://example.com/foo/bar/font-a.woff',
         endTime: 3, startTime: 1,
-        resourceType: NetworkRequest.TYPES.Font,
+        resourceType: 'Font',
       },
       {
-        url: openSansFontBold.src[0],
-        endTime: 3, startTime: 1,
-        resourceType: NetworkRequest.TYPES.Font,
+        url: 'https://example.com/foo/font-b.woff',
+        endTime: 5, startTime: 1,
+        resourceType: 'Font',
       },
-    ], webFonts)).then(result => {
-      const items = [{
-        url: openSansFontBold.src[0],
-        wastedMs: 2000,
-      }];
-      assert.strictEqual(result.rawValue, false);
-      assert.deepEqual(result.details.items, items);
-    });
-  });
+      {
+        url: 'https://example.com/foo/bar/font.woff',
+        endTime: 2, startTime: 1,
+        resourceType: 'Font',
+      },
+    ];
 
-  it('passes when all fonts have a correct font-display rule', () => {
-    const webFonts = [
-      Object.assign({}, openSansFont, {display: 'block'}),
-      Object.assign({}, openSansFontBold, {display: 'fallback'}),
+    const result = await Audit.audit(getArtifacts());
+    const items = [
+      {url: networkRecords[0].url, wastedMs: 2000},
+      {url: networkRecords[1].url, wastedMs: 3000},
+      {url: networkRecords[2].url, wastedMs: 1000},
     ];
+    assert.strictEqual(result.rawValue, false);
+    assert.deepEqual(result.details.items, items);
+  });
+
+  it('passes when all fonts have a correct font-display rule', async () => {
+    stylesheet.content = `
+      @font-face {
+        font-display: 'block';
+        src: url("./font-a.woff");
+      }
+
+      @font-face {
+        font-display: 'fallback';
+        src: url('../font-b.woff');
+      }
+
+      @font-face {
+        font-display: 'optional';
+        src: url(font.woff);
+      }
+    `;
 
-    return Audit.audit(getArtifacts([
+    networkRecords = [
       {
-        url: openSansFont.src[0],
+        url: 'https://example.com/foo/bar/font-a.woff',
         endTime: 3, startTime: 1,
-        resourceType: NetworkRequest.TYPES.Font,
+        resourceType: 'Font',
       },
       {
-        url: openSansFontBold.src[0],
-        endTime: 3, startTime: 1,
-        resourceType: NetworkRequest.TYPES.Font,
+        url: 'https://example.com/foo/font-b.woff',
+        endTime: 5, startTime: 1,
+        resourceType: 'Font',
+      },
+      {
+        url: 'https://example.com/foo/bar/font.woff',
+        endTime: 2, startTime: 1,
+        resourceType: 'Font',
       },
-    ], webFonts)).then(result => {
-      assert.strictEqual(result.rawValue, true);
-    });
+    ];
+
+    const result = await Audit.audit(getArtifacts());
+    assert.strictEqual(result.rawValue, true);
+    assert.deepEqual(result.details.items, []);
   });
```

Comment:
how do you feel about populating this with some of the `@font-face` blocks we can find out on real sites? (like the sites ward mentioned in https://github.com/GoogleChrome/lighthouse/pull/6166#issuecomment-427505006). We'd have better coverage for the regexes, plus we'd have a ready-made place to put future regression tests if we ever find we're missing weird things people are doing.

If all the examples we can find are pretty similar, could use them but vary new lines, `font-display` value, whitespace, and throw in some invalid URLs. Is there anything interesting for fonts not in the network records or not in stylesheets? (probably not for this audit)

Reply:
great idea! will do